src/inferenceql/query/permissive.cljc

```diff
@@ -52,20 +52,46 @@
                         {:event event}))))
 
 (defn identifier->variable
-  "Returns the variable node equivalent of an identifier node."
+  "Wraps an identifier node in a variable node."
   [node]
   [:variable "VAR" ws node])
 
+(defn ^:private ands->commas
+  "Replaces subsequent pairs of whitespace nodes followed by \"AND\" with
+  a comma."
+  [node]
+  ;; Really awkward to do this with a loop, but reduce/partition alternatives
+  ;; have the problem of not being able to look ahead easily, or skip over
+  ;; elements. An index is actually easier.
+  (let [cnt (count node)
+        stop-idx (dec cnt)]
+    (if (= 1 cnt)
+      node
+      (loop [i 0
+             acc []]
+        (cond (> i stop-idx) acc                            ; final pair was ws+and, so we overshot
+              (= i stop-idx) (conj acc (peek node))         ; conj last elt
+              :else (let [n1 (nth node i)
+                          n2 (nth node (inc i))]
+                      (if (and (tree/whitespace? n1)
+                               (string? n2)
+                               (re-matches #"(?i)AND" n2))
+                        (recur (+ 2 i) (conj acc ","))      ; skip to after AND
+                        (recur (inc i) (conj acc n1)))))))))
+
```

Comment:
If people think this is overkill, I can change or remove it. (Only real consequence is spaces before commas when unparsing the converted perm->strict query. E.g., `model GIVEN * EXCEPT foo AND bar AND moop` -> `model CONDITIONED BY * EXCEPT foo , bar , moop`.)

I know it's ugly, but the need to look at pairs, and skip over input makes it awkward. I considered variations using partition, reduce, rest/butlast, etc. and none of them turned out cleaner. Too many fns aren't designed to skip input or work with more than one item at a time. 

However, I was able to come up with a recursive core.match version, though, if people prefer that. Only down side is it's 50% slower.

```clojure 
(defn ands->commas2
  "Like the above, but uses core.match recursively to look at two items at a time."
  [node]
  (match/match [node]
    [([n1 n2 & more] :seq)]
    (if (and (tree/whitespace? n1)
             (string? n2)
             (re-matches #"(?i)AND" n2))
      (cons "," (ands->commas2 more))
      (cons n1 (ands->commas2 (cons n2 more))))

    :else node))
```

I also tried using `into []` and vectors, but lists are faster for building piece-wise.